File: extract.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

const docxPath = path.join(__dirname, 'docx');
const outPath = path.join(__dirname, 'pdf_extracted.txt');

async function extract() {
    const files = fs.readdirSync(docxPath).filter(f => f.endsWith('.pdf'));
    let combinedText = '';

    for (const file of files) {
        combinedText += `\n\n--- Start of ${file} ---\n\n`;
        const dataBuffer = fs.readFileSync(path.join(docxPath, file));
        try {
            const data = await pdf(dataBuffer);
            combinedText += data.text;
        } catch (e) {
            combinedText += `Error extracting ${file}: ${e.message}\n`;
        }
        combinedText += `\n\n--- End of ${file} ---\n\n`;
    }

    fs.writeFileSync(outPath, combinedText);
    console.log('Extraction complete. Saved to pdf_extracted.txt');
}

extract();

File: extract.py
import os
import sys
import subprocess

def install(package):
    subprocess.check_call([sys.executable, "-m", "pip", "install", package])

try:
    import PyPDF2
except ImportError:
    install('PyPDF2')
    import PyPDF2

pdf_dir = 'docx'
with open('pdf_extracted.txt', 'w', encoding='utf-8') as out_file:
    for filename in os.listdir(pdf_dir):
        if filename.endswith('.pdf'):
            out_file.write(f'\n\n--- Start of {filename} ---\n\n')
            try:
                reader = PyPDF2.PdfReader(os.path.join(pdf_dir, filename))
                for page in reader.pages:
                    text = page.extract_text()
                    if text:
                        out_file.write(text + '\n')
            except Exception as e:
                out_file.write(f'Error: {e}\n')
            out_file.write(f'\n\n--- End of {filename} ---\n\n')
print("Done extracting PDFs")
